kroxylicious/src/main/java/io/kroxylicious/proxy/classloader/ClassloaderUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.classloader;
+
+import java.util.Optional;
+
+public class ClassloaderUtils {
+    public static <T> Optional<Class<? extends T>> loadClassIfPresent(Class<T> superClass, String binderClass) {
+        return loadClass(superClass, binderClass, false);
+    }
+
+    public static <T> Class<? extends T> loadRequiredClass(Class<T> superClass, String configurationHookClass) {
+        return loadClass(superClass, configurationHookClass, true)
+                .orElseThrow(() -> new IllegalArgumentException("could not load " + configurationHookClass));
+    }
+
+    public static <T> Optional<Class<? extends T>> loadClass(Class<T> superClass, String configurationHookClass, boolean failIfNotFound) {
```

Comment:
`configurationHookClass` -> `requestedClass` given this isn't specific to configuration hooks.

Reply:
fixed the naming in this class